Library/Homebrew/system_command.rb

```diff
@@ -77,6 +78,15 @@ def command
     [*sudo_prefix, *env_args, executable.to_s, *expanded_args]
   end
 
+  def redacted_command
+    command.map do |arg|
+      @secrets.each do |secret|
+        arg = arg.gsub secret, "REDACTED"
```

Comment:
Can this be `********` instead? Also, might be nice to use `inject` here or similar to modify the arg in place.

Reply:
> Can this be ******** instead? 

There is a `.shelljoin` applied to the `redacted_command`, it will print out `\*\*\*\*`. So I choose to use `REDACTED` here.

> Also, might be nice to use inject here or similar to modify the arg in place.

I am not sure how to use inject. Can you show me an example? The replacement is not in place because the arg may be a freezed string.

Reply:
I think it would be good to resolve that regardless; the standard practise for such things is `********` and `REDACTED` is pretty ugly.



> I am not sure how to use inject.

Use the documentation @sjackman noted and pass the `arg` which is modified each time (`dup`d once initially if needed and then `freeze` when returned.

Reply:
I have updated PR to print `******`.  And I slightly updated the code to use in-place replacement.

As for `inject`, I still don't see how it can be used here and why it should be used.

Reply:
@xu-cheng Pass the `arg` string as the argument rather than assigning back to it. It's more idiomatic Ruby.

Reply:
Would you mind to show me the actual code example? I am sorry that what is on your mind escapes me.

Reply:
```ruby
command.map do |arg|
  @secrets.inject(arg.dup) do |arg, secret|
    arg.gsub! secret, "******"
  end.freeze
end
```

Reply:
Thanks

Reply:
@MikeMcQuaid Unfortunately, the above code does not work:
```ruby
command = %w[A B]
secrets = %w[B]
result =  command.map do |arg|
  secrets.inject(arg.dup) do |arg, secret|
    arg.gsub! secret, "******"
  end.freeze
end 
p result # we get: [nil, "******"]
```

This following works, but it is not in-place replacement.
```ruby
command = %w[A B]
secrets = %w[B]
result =  command.map do |arg|
  secrets.inject(arg) do |arg, secret|
    arg.gsub secret, "******"
  end.freeze
end 
p result # we get: ["A", "******"]
```

Reply:
@xu-cheng Cool, second seems fine then, thanks.